includes/admin/extensions/abstract-extension.php

```diff
@@ -0,0 +1,302 @@
+<?php
+
+namespace EDD\Admin\Extensions;
+
+use \EDD\Admin\Pass_Manager;
+
+abstract class Extension {
+
+	/**
+	 * The product ID. This only needs to be set if the extending class is
+	 * for a single product.
+	 *
+	 * @since 2.11.x
+	 * @var int
+	 */
+	protected $item_id;
+
+	/**
+	 * The product data.
+	 *
+	 * @var boolean
+	 */
+	private $product_data = false;
```

Comment:
```suggestion
	protected $product_data = false;
```

`private` means only this abstract class can access it. Because of that, each of the child classes are actually dynamically declaring a new public property. Changing to `protected` allows them to access it properly.

Reply:
Actually building on this, I'm not quite sure I like how this change sets the product data in the constructor. Example: https://github.com/awesomemotive/easy-digital-downloads/blob/try/product-education/includes/admin/extensions/product-education/class-email-marketing.php#L20-L23

I think it would actually be really nice to only run the API requests when absolutely needed. (Meaning: both the cache has expired AND you're currently viewing the relevant settings page.) Right now this will be doing API requests if the cache has expired and you're viewing ANY admin page at all. That's bad for both the customer and for us.